module_24_Python/1_taxes.py

```diff
@@ -0,0 +1,148 @@
+и производных от него классов Apartment, Car и CountryHouse.
+Базовый класс должен иметь атрибут worth (стоимость), который передаётся в конструктор,
+и метод расчёта налога,
+переопределённый в каждом из производных классов.
+Налог на квартиру вычисляется как 1/1000 её стоимости, на машину — 1/200, на дачу — 1/500.
+Каждый дочерний класс должен иметь конструктор с одним параметром,
+передающий свой параметр конструктору базового класса.
+
+Разработайте интерфейс программы.
+Пусть она запрашивает у пользователя количество его денег и стоимость имущества,
+а затем выдаёт налог на соответствующее имущество и показывает,
+сколько денег ему не хватает (если это так).
+"""
+
+
+class Property:
+    """
+    class Property contains property worth
+
+    Args:
+        worth (float): worth of property
+        Attributes:
+            tax_rate (int): tax rate
+
+    """
+
+    tax_rate = 1000
+
+    def __init__(self, worth: float):
+        self._worth = worth
+
+    def get_worth(self):
+        """
+        Get worth
+
+        :return: _worth
+        :rtype: float
+        """
+        return self._worth
+
+    def get_tax(self):
+        """
+        Get tax
+
+        :return: tax
+        :rtype: float
+        """
+        return round(self._worth / self.tax_rate, 2)
+
+    def __str__(self):
+        return f'Worth of property: {self._worth}'
+
+
+class Apartment(Property):
+    """
+    Class Apartment
+
+    Args:
+        worth (float): worth of property
+        Attributes:
+            tax_rate (int): = 1000
+
+    """
+
+    tax_rate = 1000
+
+    def __init__(self, worth: float):
+        super().__init__(worth)
+
+    def get_tax(self):
+        """
+        Get tax
+
+        :return: tax
+        :rtype: float
+        """
+        return round(self.get_worth() / self.tax_rate, 2)
+
+
+class Car(Property):
+    """
+        Class Car
+        tax_rate = 200
+
+    """
+
+    tax_rate = 200
+
+    def __init__(self, worth: float):
+        super().__init__(worth)
+
+    def get_tax(self):
+        """
+            Get tax
+
+            :return: tax
+            :rtype: float
+        """
+        return round(self.get_worth() / self.tax_rate, 2)
+
+
+class CountryHouse(Property):
+    """
+        Class CountryHouse
+        tax_rate = 1000
+
+    """
+
+    tax_rate = 500
+
+    def __init__(self, worth: float):
+        super().__init__(worth)
+
+    def get_tax(self):
+        """
+            Get tax
+
+            :return: tax
+            :rtype: float
+        """
+        return round(self.get_worth() / self.tax_rate, 2)
+
+
+person_money = float(input('Amount of your money: '))
+# person_properties = dict()
+apartment = Apartment(float(input('Worth of your apartment: ')))
+car = Car(float(input('Worth of your car: ')))
+country_house = CountryHouse(float(input('Worth of your country house: ')))
+
+person_properties = {per_property: per_property.get_tax() for per_property in [apartment, car, country_house]}
```

Comment:
The word `per` usually stands for `for`
Rename `per_property` -> `property` (or `prop`)

Reply:
done